src/ui/actions/session.ts

```diff
@@ -165,31 +166,51 @@ export function createSocket(
 
       const loadPoint = new URL(window.location.href).searchParams.get("point") || undefined;
 
+      const queuedActions: Array<Action | UIThunkAction> = [];
+      let flushTimeoutId: NodeJS.Timeout | null = null;
+
+      function flushQueuedActions() {
+        flushTimeoutId = null;
+
+        const actions = queuedActions.splice(0);
```

Comment:
:question: I'm confused by the `.splice(0)`. What's that doing?

Reply:
Grabbing all of the current actions _and_ clearing out the queue so we can add more actions to it later.

```js
const actions = queuedActions.splice(0);

// actions now has all of the items queuedActions used to have
// queuedActions is now empty
```

Reply:
Ah. I probably woulda just replaced the existing array or something :)